src/main/java/com/microsoft/sqlserver/jdbc/SQLServerADAL4JUtils.java

```diff
@@ -41,25 +41,30 @@ static SqlFedAuthToken getSqlFedAuthToken(SqlFedAuthInfo fedAuthInfo, String use
             return new SqlFedAuthToken(authenticationResult.getAccessToken(), authenticationResult.getExpiresOnDate());
         } catch (MalformedURLException | InterruptedException e) {
             throw new SQLServerException(e.getMessage(), e);
-        } catch (ExecutionException e) {
+        } catch (Exception e) {
```

Comment:
Blanket catching all exceptions is supposed to be a code smells. Is there a way for us to not do this?